agent/replacecall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -0,0 +1,398 @@
+      }
+    }
+    cg.update();
+
+    debug_transform.log("transformClass: EXIT %s", cg.getClassName());
+    return transformed;
+  }
+
+  /**
+   * Transforms the specified method to replace mapped calls.
+   *
+   * @param mg the method generator
+   * @param ifact the instrument factory for the enclosing class of this method
+   * @param pgen the constant pool for this class
+   * @return true if the method was modified, false otherwise
+   */
+  private boolean transformMethod(MethodGen mg, InstructionFactory ifact, ConstantPoolGen pgen)
+      throws IllegalClassFormatException {
+    boolean transformed = false;
+    // Loop through each instruction, making substitutions
+    debug_transform.log("transformMethod: ENTER %s.%s%n", mg.getClassName(), mg.getName());
+    InstructionList instructionList = mg.getInstructionList();
+    InstructionHandle instructionHandle = instructionList.getStart();
+    while (instructionHandle != null) {
+
+      debug_instrument_inst.log(
+          "%s.%s: instrumenting instruction %s%n",
+          mg.getClassName(), mg.getName(), instructionHandle);
+
+      // The next instruction for next iteration
+      InstructionHandle nextHandle = instructionHandle.getNext();
+
+      InvokeInstruction instruction =
+          getReplacementInstruction(mg, instructionHandle.getInstruction(), ifact, pgen);
+      debug_instrument_inst.log(
+          " %s.%s new inst: %s%n", mg.getClassName(), mg.getName(), instruction);
+
+      if (instruction != null) {
+        transformed = true;
+        instructionHandle.setInstruction(instruction);
+      }
+
+      instructionHandle = nextHandle;
+    }
+    debug_transform.log("transformMethod: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+    return transformed;
+  }
+
+  /**
+   * Returns the instruction to call a replacement method instead of the original method.
+   *
+   * @param mg the BCEL representation of the method being transformed
+   * @param inst the instruction to replace
+   * @param ifact the instruction factory for the enclosing class
+   * @return the new instruction, or null if the instruction has no replacement
+   */
+  private InvokeInstruction getReplacementInstruction(
+      MethodGen mg, Instruction inst, InstructionFactory ifact, ConstantPoolGen pgen)
+      throws IllegalClassFormatException {
+    debug_transform.log(
+        "getReplacementInstruction: ENTER %s.%s%n", mg.getClassName(), mg.getName());
+    if (!(inst instanceof InvokeInstruction)) {
+      debug_transform.log(
+          "getReplacementInstruction: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+      return null;
+    }
+
+    InvokeInstruction invocation = (InvokeInstruction) inst;
+    MethodSignature orig = MethodSignature.of(invocation, pgen);
+
+    MethodSignature call = replacementMap.get(orig);
+    if (call == null) {
+      debug_transform.log(
+          "%s.%s: No replacement for %s%n", mg.getClassName(), mg.getName(), orig.toString());
```

Comment:
This terse message is ambiguous about whether no replacement was needed, or a replacement was needed and it's a problem that none was found.  Clarify this, please.